hedera-node/src/main/java/com/hedera/services/utils/UnzipUtility.java

```diff
@@ -58,24 +58,28 @@ private static void extractSingleFile(ZipInputStream inputStream, String filePat
 		}
 	}
 
-	public static void unzip(byte[] bytes, String dstDirectory) throws IOException {
-		File destDir = new File(dstDirectory);
+	public static void unzip(final byte[] bytes, final String dstDirectory) throws IOException {
+		final File destDir = new File(dstDirectory);
 		if (!destDir.exists()) {
 			log.info("create dir " + destDir);
 			destDir.mkdir();
 		}
-		ZipInputStream zipIn = new ZipInputStream(new ByteArrayInputStream(bytes));
+		final ZipInputStream zipIn = new ZipInputStream(new ByteArrayInputStream(bytes));
 		ZipEntry entry = zipIn.getNextEntry();
 		while (entry != null) {
 			String filePath = dstDirectory + File.separator + entry.getName();
+			final File fileOrDir = new File(filePath);
+			filePath = fileOrDir.getCanonicalPath();
+			if(!filePath.startsWith(destDir.getCanonicalPath() + File.separator)) {
+				throw new IllegalArgumentException("Zip file entry has invalid path " + filePath +
+						". Only specify a filePath with correct canonical path name.");
+			}
 			if (!entry.isDirectory()) {
-				File newFile = new File(filePath);
 				log.info("create file {}", filePath);
 				extractSingleFile(zipIn, filePath);
```

Comment:
need test

Reply:
The test cases to cover these lines are in MerkleAccountPropertyTest